discovery/src/lib.rs

```diff
@@ -111,8 +111,7 @@ impl Builder {
                     None,
                     port,
                     &["VERSION=1.0", "CPath=/"],
-                )
-                .unwrap();
+                ).map_err(|e| Error::DnsSdError(io::Error::new(io::ErrorKind::Unsupported, e)))?;
```

Comment:
I can either:

- Change this to `ErrorKind::Other`.
- Use a macro to use `ErrorKind::Other` only on older Rusts (very ugly!).
- Bump the MSRV.

This PR currently implements the third option, let me know if you'd prefer another.